modules/apps/item-selector/item-selector-taglib/src/main/resources/META-INF/resources/image_selector/js/ImageSelector.js

```diff
@@ -51,16 +53,21 @@ const ImageSelector = ({
 }) => {
 	const [errorMessage, setErrorMessage] = useState('');
 	const [fileName, setFileName] = useState('');
-	const [image, setImage] = useState({
-		fileEntryId,
-		src: imageURL,
-	});
 	const [imageCropRegion, setImageCropRegion] = useState(
 		initialImageCropRegion
 	);
 	const [progressData, setProgressData] = useState();
 	const [progressValue, setProgressValue] = useState(0);
 
+	const [image, setImage] = useLiferayState(imageSelectorCoverImageAtom);
+
+	useEffect(() => {
+		setImage({
+			fileEntryId,
+			src: imageURL,
+		});
+	}, [fileEntryId, imageURL, setImage]);
```

Comment:
As noted in [the original draft PR](https://github.com/liferay-frontend/liferay-portal/pull/970), there is a bit of architectural weirdness here in the way `blogs-web` and `item-selector-taglib` work together, and it comes to the surface here.

Specifically, they are supposedly modular and independent (eg. `item-selector-taglib` is also used by Commerce, via the `<liferay-item-selector:image-selector>`), but the reality is that `item-selector-taglib` is secretly entangled with `blogs-web` because it fired these `coverImage` events behind the scenes. And because the events were global and not keyed by any kind of identifier, `blogs-web` is assuming there is only one "cover image" image selector displayed at any given moment. (Gah, as I test this manually now, I see we _still_ have things entangled here... if you change the small image, it changes the cover image too... going to move this into draft and re-test it...)

Anyway, this coupling shows up here where we have to "abuse" `useEffect` to set the initial image based on our props. In a more "normal" situation, something that renders the `ImageSelector` would set up the global state from the outside, before rendering, because it really would be global state as opposed to some kind of secret back-channel.

Reply:
Final version of the pull avoids _some_ of this weirdness in the course of fixing the caption bug. We had to go back to `useState` for managing the image itself, and the `useEffect` ends up just calling `State.write` to communicate with `blogs-web`. So the "render then update" is gone, and we're not dealing with the situation of wishing that "the global state [was set up] from the outside, before rendering".